internal/integration/integration_test.go

```diff
@@ -19,7 +19,8 @@ func TestWalkingSkeleton(t *testing.T) {
 		GraphQL: api.NewClient(cfg.BuildkiteToken),
 	}.Init()
 	ctx := context.Background()
-	pipelineID := tc.CreatePipeline(ctx)
+	pipelineID, cleanup := tc.CreatePipeline(ctx)
+	defer cleanup()
```

Comment:
these could potentially be swapped out for `t.Cleanup()` calls, but the actual distinction and when cleanup works better than defer escapes me... regardless, if this works fine then i probably wouldn't futz with it too much

Reply:
Oh, TIL `t.Cleanup`. Apparently, `defer` can be buggy if you invoke `t.Run`, which a `t.Parallel` inside a test. The main test function can return (and run the defer) while the function is `t.Run` is still running 😱

We don't use `t.Run` in the integration tests, but I think we should still use `t.Cleanup` in case we do in the future and get puzzled by the flake.

Reply:
Source: https://brandur.org/fragments/go-prefer-t-cleanup-with-parallel-subtests